pylibftdi/device.py

```diff
@@ -543,24 +546,28 @@ def readline(self, size=0):
         This only works for mode='t' on Python3
         """
         lsl = len(os.linesep)
-        line_buffer = []
+        line_buffer: list[str] = []
         while True:
             next_char = self.read(1)
             if next_char == '' or (0 < size < len(line_buffer)):
                 break
+            if isinstance(next_char, bytes):
+                next_char = self.decoder.decode(next_char)
```

Comment:
Not tested, but I'm not sure this will work for e.g. utf-8 data where decoding a single byte doesn't always make sense. As line 546 implies, this is only intended for things where `read()` will already return a `str` rather than `bytes` (apart from anything else, if bytes are being returned, the `next_char == ''` check on line 552 will never trigger and EOF won't end the attempts to read things...)

Might be better to check for a text encoding (or at simplest, do `assert isinstance(next_char, str)` after the `read()`)